deep_field_metadetect/gaussmom/gaussmom.py

```diff
@@ -0,0 +1,558 @@
+            res_flags,
+        )
+
+    res_e, res_e_err, res_e_cov, res_flags = jax.lax.cond(
+        res_flags == 0,
+        valid_flags_fn,
+        invalid_flags_fn,
+        operand=None,
+    )
+    res_e1 = res_e[0]
+    res_e2 = res_e[1]
+
+    pars = jnp.array(
+        [
+            sums[mv_ind],
+            sums[mu_ind],
+            res_e1,
+            res_e2,
+            res_T,
+            res_flux,
+        ]
+    )
+
+    res = GaussMomData(
+        obs=gaussmom_obs,
+        npix=npix,
+        flags=res_flags,
+        wsum=sums_norm,
+        flux=res_flux,
+        sums=res_sums,
+        sums_cov=res_sums_cov,
+        sums_norm=res_sums_norm,
+        flux_flags=res_flux_flags,
+        T_flags=res_T_flags,
+        flux_err=res_flux_err,
+        T=res_T,
+        T_err=res_T_err,
+        s2n=res_s2n,
+        e1=res_e1,
+        e2=res_e2,
+        e=res_e,
+        e_err=res_e_err,
+        e_cov=res_e_cov,
+        sums_err=res_sums_err,
+        pars=pars,
+    )
+
+    # _add_moments_by_name(res) # shouldn't modify sums or sums_cov
+
+    # TODO: not yet computing the normalized momemnts [Mv, Mu, M1, M2, MT, MF]
+
+    return res
+
+
+def _add_moments_by_name(res):  # TODO
+    sums = res["sums"]
+    sums_cov = res["sums_cov"]
+
+    mf_ind = MOMENTS_NAME_MAP["MF"]
+    fsum = sums[mf_ind]
+    fsum_err = np.sqrt(sums_cov[mf_ind, mf_ind])
+
+    # add in named sums normalized by flux sum (weight * image).sum()
+    # we don't store flags or errors for these
+    mkeys = list(MOMENTS_NAME_MAP.keys())
+    for name in mkeys:
+        ind = MOMENTS_NAME_MAP[name]
+        if ind > sums.size - 1:
+            continue
+
+        err_name = f"{name}_err"
+
+        if name in ["MF", "M00"]:
+            res[name] = fsum
+            res[err_name] = fsum_err
+        else:
+            if fsum > 0:
+                res[name] = sums[ind] / fsum
+                res[err_name] = get_ratio_error(
+                    sums[ind],
+                    sums[mf_ind],
+                    sums_cov[ind, ind],
+                    sums_cov[mf_ind, mf_ind],
+                    sums_cov[ind, mf_ind],
+                )
+            else:
+                res[name] = np.nan
+                res[err_name] = np.nan
+
+
+@jax.jit
+def get_ratio_error(a, b, var_a, var_b, cov_ab):
+    """
+    Compute the error on the ratio a / b using JAX.
+    """
+    var = get_ratio_var(a, b, var_a, var_b, cov_ab)
+
+    var = jnp.clip(var, 0.0, jnp.inf)
+    error = jnp.sqrt(var)
+    return error
+
+
+@jax.jit
+def get_ratio_var(a, b, var_a, var_b, cov_ab):
+    """
+    Compute the variance of (a/b).
+    """
+
+    # Ensure no division by zero
+    b = jnp.where(
+        b == 0, 1e-100, b
+    )  # TODO: This does not raise a value error like ngmix
+
+    rsq = (a / b) ** 2
+
+    var = rsq * (var_a / a**2 + var_b / b**2 - 2 * cov_ab / (a * b))
+
+    # var = jnp.where(var>1e100, jnp.inf, var)
```

Comment:
Did you decide against using jnp.inf? What happened here?

Reply:
This line does not exist in original ngmix. If b==0 ngmix raises a ValueError but to make it jitable, I set b to a very small value. I intoduced this line to make sure that var is retuned inf it is it too large.
I commented it out just to discuss if a better way to go about this

Reply:
So my general approach has been to insert NaNs when something catastrophic happens. equinox has APIs for raising errors inside JITed code, but I have not gone down that route. So I'd insert a NaN here for now.